### ISSUE TITLE
Exact Marginalization??

### ISSUE THREAD

**Issue by wlj6816** [harvardnlp/var-attn#3]
It is confusing me that categorical attention with exact evidence sets  inference_network_type not Nan.
In my option, categorical attention with exact evidence(Marginal Likelihood) is as follow:
 
![image](https://user-images.githubusercontent.com/32925120/50441226-b28c4f80-0934-11e9-9262-f35d688f0919.png)
the inference network produces the parameter of q(z) of variational attention.

**Comment by da03** [comment]
Yeah, it was just a placeholder, q is actually not being used in that mode.

**Comment by wlj6816** [comment]
> Yeah, it was just a placeholder, q is actually not being used in that mode.

Thank you very much. I am working on a similar task as your paper and need to compare our performance with your method. But you do not apply any of the commonly-utilized techniques to improve performance on VQA like "Bottom-Up and Top-Down Attention for Image Captioning and Visual Question Answering". I need some details of the experiences about VQA.
Could I get the code of VQA for this paper? Can I add your WeChat?
